tools/codegen/dest/register_dispatch_key.py

```diff
@@ -232,7 +232,7 @@ def gen_class_set_output(self, k: SchemaKind, parent_class: str, generate_super:
 """
 
     def gen_class_set_output_body(self, k: SchemaKind) -> str:
-        if self.dispatch_key in [DispatchKey.CUDA, DispatchKey.CompositeExplicitAutograd]:
+        if self.backend_index.dispatch_key in [DispatchKey.CUDA, DispatchKey.CompositeExplicitAutograd]:
```

Comment:
we could in theory also store stuff like "needs device guard" in the backend index, but maybe that's overkill.

Reply:
yeah, that's what I meant 😃. Sounds good.